lib/config/options.rb

```diff
@@ -83,17 +83,7 @@ def reload_from_files(*files)
     end
 
     def to_hash
-      result = {}
-      marshal_dump.each do |k, v|
-        if v.instance_of? Config::Options
-          result[k] = v.to_hash
-        elsif v.instance_of? Array
-          result[k] = descend_array(v)
-        else
-          result[k] = v
-        end
-      end
-      result
+      __convert_to_hash(marshal_dump)
```

Comment:
👌 , removed.

Yes, I am running this code in Rails, but I'm not sure about the connection to DeepMerge. If I take the example from this spec and drop into a debugger at https://github.com/railsconfig/config/blob/master/lib/config/options.rb#L110:

```
[106, 115] in /Users/cabeer/Projects/3rd-party/config/lib/config/options.rb
   106:     end
   107: 
   108:     def merge!(hash)
   109:       require 'byebug'; byebug
   110:       current = to_hash
=> 111:       DeepMerge.deep_merge!(hash, current)
   112:       marshal_load(__convert(current).marshal_dump)
   113:       self
   114:     end
   115: 
```

The two hashes have different keys (strings vs symbols):
```
(byebug) hash
{
  [...]
  "size"=>3
  "inner"=>{
     "something1"=>"new blah"
  }
}
```

```
(byebug) current
{
  :size=>1,
  :inner=>{
    :something1=>"blah1",
    :something2=>"blah2"
   },
   [...]
  ...
}
```

And the merge (dutifully) merges them into an even bigger hash:

```
(byebug) current
{
  :size=>1, 
  :inner=>{:something1=>"blah1", :something2=>"blah2"},
  [...]
   "size"=>"3",
   "inner"=>{"something1"=>"new blah"}
}
```

With simple config structures (like `size` above), this isn't a problem when we load the hash back into OpenStruct; the last key (which happens to contain the data we want) wins.

With a complex structure (like `inner`), the last element replaces the value entirely, removing the previous settings.